libcloud/test/compute/test_ecs.py

```diff
@@ -535,27 +535,29 @@ def test_ex_list_security_groups(self):
         self.assertEqual('2015-06-26T08:35:30Z', sg.creation_time)
 
     def test_ex_join_security_group(self):
-        ex_security_group_id_value = 'sg-28ou0f3xa'
+        self.sg_id = "sg-28ou0f3xa"
         result = self.driver.ex_join_security_group(self.fake_node,
-                                                    group_id=ex_security_group_id_value)
+                                                    group_id=self.sg_id)
```

Comment:
Can you please explain why you switched to storing `sg_id` inside `self` ?

Reply:
You can see, ```sg_id``` is needed by five functions (maybe more in the future) of ```ecs.py```.  We should ```set``` it in ```test_``` functions and check it in class ```ECSMockHttp```. Storing ```sg_id``` inside ```self``` make it easier for us to get the value in ```ECSMockHttp``` and keep consistent in other future ```test_``` functions.

I think it's better to treat it as a attribute of class ```security_group```, but ```libcloud/compute/base.py``` doesn't include that class. It is also okay to leave it as ```ex_security_group_id_value``` and check it with the actual value.

Reply:
I see! But you need to do that in the `setUp` method. It's clearer and means the tests still pass if you run them with a different order with say `pytest --failed-first` or if you run only one of them with `pytest -k`.

Reply:
Great idea.